public/components/event_analytics/utils/utils.tsx

```diff
@@ -301,6 +310,61 @@ export const getDefaultVisConfig = (statsToken: statsChunk) => {
   };
 };
 
+export const fetchOtelMetric = async ({
+  visualizationName,
+  startTime,
+  endTime,
+  setIsError,
+  setIsLoading,
+}: {
+  visualizationName: string;
+  startTime: string;
+  endTime: string;
+  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>;
+  setIsError: React.Dispatch<React.SetStateAction<VizContainerError>>;
+}) => {
+  console.log('visualizationName: ', visualizationName);
+  const osdHttp = getOSDHttp();
+  const indexAndDocumentName = extractIndexAndDocumentName(visualizationName);
+  const index = indexAndDocumentName[0];
+  const documentName = indexAndDocumentName[1];
+
+  const fetchSampleDocument = await fetchSampleOTDocument(index, osdHttp, documentName)();
+  const source = fetchSampleDocument.hits[0]._source;
+
+  const dataBinsPromises = source.buckets.map(async (bucket: any) => {
+    try {
+      const formattedStartTime = convertDateTime(startTime, false, false, false, true);
+      const formattedEndTime = convertDateTime(endTime, false, false, false, true);
+      const fetchingAggregatedBinCount = await fetchAggregatedBinCount(
+        bucket.min.toString(),
+        bucket.max.toString(),
+        formattedStartTime,
+        formattedEndTime,
+        documentName,
+        osdHttp,
+        index,
+        setIsError,
+        setIsLoading
+      )();
+
+      return {
+        xAxis: bucket.min + ' - ' + bucket.max,
+        'count()': fetchingAggregatedBinCount?.nested_buckets?.bucket_range?.bucket_count?.value,
+      };
+    } catch (error) {
+      console.error('Error processing bucket:', error);
+      return null;
+    }
+  });
+  const jsonData = await Promise.all(dataBinsPromises);
+  const formatDataBinsName = () => {
+    return { dataBins: { jsonData } };
+  };
+  console.log(formatDataBinsName());
+  return formatDataBinsName().dataBins;
```

Comment:
same question here, is it just
```suggestion
  const jsonData = await Promise.all(dataBinsPromises);
  return { jsonData };
```